src/client/systems/NetworkingClient.cpp

```diff
@@ -263,14 +259,33 @@ void NetworkingClient::unreliableRecv() {
         if (status != sf::Socket::NotReady) {
             GameState game_state;
             packet >> game_state;
-            std::lock_guard<std::mutex> m_gameState_guard(m_gameState_lock);
-            m_gameState_t = game_state;
+            updateGameStateBuffer(game_state);
         }
 
         std::this_thread::sleep_for(GAME_SETTINGS.CLIENT_UNRELIABLE_RECV_SLEEP);
     }
 }
 
+void NetworkingClient::updateGameStateBuffer(GameState& game_state) {
+    std::lock_guard<std::mutex> m_gameStateBuffer_guard(m_gameStateBuffer_lock);
+    std::lock_guard<std::mutex> m_serverGameStateMetric_guard(m_serverGameStateMetric_lock);
+    m_gameStateBuffer_t[game_state.core.tick] = game_state;
+    if (m_gameStateBuffer_t.size() > GAME_SETTINGS.CLIENT_GAME_STATE_BUFFER_SIZE) {
+        if (m_gameStateBuffer_t.begin()->first != game_state.core.tick) {
```

Comment:
I dont understand this code, it's comparing the first entry in the game state buffer's tick to the the update's tick. if they aren't equal it updates the serverGameStateMetric. I guess im confused what serverGameStateMetric is supposed to be without grepping around. might be worth adding a comment explaining what's going on

Reply:
I dont get why it's erasing the first entry just because it doesnt equal the game state update we've received

Reply:
ie I thought the behavior would be that the smallest tick would be removed from the game state buffer *if* it was smaller than the incoming game state update's tick. if it's larger, than I thought the update would be discarded.

Reply:
This code got a bit more complicated because I wanted to track when we put something new into the buffer for `m_serverGameStateMetric`

But essentially what's happening is 
272: we insert the new game state
273-277: if the exceeded our size then we delete the smallest member, if the smallest member isn't the think we just entered then it's a new game state (that won't be removed) so update the count.